tests/sals/nginx/test_nginx_sal.py

```diff
@@ -0,0 +1,249 @@
+from jumpscale.loader import j
+from unittest import TestCase
+from time import time
+from jumpscale.sals.nginx.nginx import LocationType
+import requests
+import os
+from subprocess import Popen
+import math
+from jumpscale.sals.nginx.nginx import PORTS
+
+HOST = "127.0.0.1"
+NGINX_CONFIG_FILE = "nginx.conf"
+
+
+def wait_connection_lost_test(url, timeout):
+    start_time = time()
+    while time() - start_time <= timeout:
+        if not j.sals.nettools.wait_http_test(url, math.ceil(time() - start_time)):
+            return True
+    return False
+
+
+def request_content(url, verify=False):
+    return requests.get(url, verify=verify).content.decode()
+
+
+class HTTPServer:
+    def __init__(self):
+        self.process = None
+
+    def run(self, port=9090):
+        devnull = open(os.devnull, "wb")
+        self.process = Popen(["python", "-m", "http.server", str(port)], stdout=devnull, stderr=devnull)
+
+    def stop(self):
+        if self.process is None:
+            raise RuntimeError("The server is not running.")
+        self.process.terminate()
+        self.process = None
+
+
+class TestNginxSal(TestCase):
+    def setUp(self):
+        self.nginx_conf = j.sals.nginx.get("testing")
+        self.config_base_dir = self.nginx_conf.cfg_dir
+        self.config_file = self.nginx_conf.cfg_file
+        self.logs_dir = self.nginx_conf._cfg_dir
+        self.nginx_server = j.tools.nginx.get("testing_server", server_name="testing")
+
+    def test01_basic(self):
+        """ # Test case for server initialization with default config
```

Comment:
Can we follow the shape of the test docstring? 
https://github.com/threefoldtech/js-sdk/blob/development/tests/tools/restic/test_restic.py#L159-L167